modules/views/components/civicrm.core.inc

```diff
@@ -2911,7 +2911,7 @@ function _civicrm_core_data(&$data, $enabled) {
 
   while ($dao->fetch()) {
     // call getTree using $dao->id as groupID, $dao->extends as entityType, with possible subtypes in $dao->extends_entity_column_value
-    $extendsContactSubtype = (('Contact' === $dao->extends) && !empty($dao->extends_entity_column_value));
+    $extendsContactSubtype = (in_array($dao->extends, array('Contact', 'Individual', 'Organization', 'Household')) && !empty($dao->extends_entity_column_value));
```

Comment:
The patch, and this proposed alteration does not fix the issue for custom fields assigned to Relationships. I'll take a look at an alternative solution.